src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToRfc3339Transformer.php

```diff
@@ -32,15 +39,25 @@ public function transform($dateTime)
         }
 
         if ($this->inputTimezone !== $this->outputTimezone) {
-            $dateTime = clone $dateTime;
+            if (!$dateTime instanceof \DateTimeImmutable) {
+                $dateTime = clone $dateTime;
+            }
```

Comment:
@xabbuh technically, nothing was changing the input in the proposal. A DateTimeImmutable is immutable.